File: app/src/main/java/seigneur/gauvain/mycourt/utils/ConnectivityReceiver.java
package seigneur.gauvain.mycourt.utils;

import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class ConnectivityReceiver {

    ConnectivityManager connectivityManager;
    boolean isOnline;

    public ConnectivityReceiver(ConnectivityManager connectivityManager) {
        this.connectivityManager=connectivityManager;
    }

    public boolean isOnline() {
        NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo();
        //should check null because in airplane mode it will be null
        isOnline= (activeNetwork != null && activeNetwork.isConnectedOrConnecting());
        return isOnline;

    }

}

File: app/src/main/java/seigneur/gauvain/mycourt/utils/ConnectivityReceiver.kt
package seigneur.gauvain.mycourt.utils

import android.net.ConnectivityManager
import android.net.NetworkInfo

class ConnectivityReceiver(private var connectivityManager: ConnectivityManager) {
    private var isOnline: Boolean = false

    fun isOnline(): Boolean {
        val activeNetwork = connectivityManager.activeNetworkInfo
        //should check null because in airplane mode it will be null
        isOnline = activeNetwork != null && activeNetwork.isConnectedOrConnecting
        return isOnline

    }

}